src/lib.rs

```diff
@@ -57,19 +62,50 @@ enum VadState {
     Silence,
 }
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[derive(Clone, Debug)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum VadTransition {
     SpeechStart {
         /// When the speech started, in milliseconds since the start of the VAD session.
         timestamp_ms: usize,
     },
     SpeechEnd {
+        /// When the speech started, in milliseconds since the start of the VAD session.
+        start_timestamp_ms: usize,
         /// When the speech ended, in milliseconds since the start of the VAD session.
-        timestamp_ms: usize,
+        end_timestamp_ms: usize,
+        /// The active speech samples. This field is skipped in serde output even serde feature is enabled.
+        #[cfg_attr(feature = "serde", serde(skip))]
```

Comment:
Question: so now if we serialize a `VadTransition::SpeechEnd` can we deserialize it or will it error because there's nothing like `serde(default)` or similar? Serde has an annoying thing where it can generate code that ends up erroring at runtime too easily :eyes: